### PR TITLE
glib and friends: migrate to python@3.10

### PR DESCRIPTION
**Pull request by carlocab**
Let's hope I don't need to include more in this PR...

### CODE REVIEW

**Comment by carlocab** [comment]
I count about ~40 formulae here where the Python dependency comes from Python bindings for a C/C++ application/library.

@Homebrew/core thoughts about making these dependencies on Python `:build` and `:test` only? That will allow me to split them into separate single-formula PRs, making things easier on CI (and future migrations easier). We can add `caveats` recommending a `brew install python@3.x` to use the Python bindings if we're concerned about UX.

**Comment by carlocab** [comment]
Here are candidates for separate migration by changing Python to a build/test dependency. Feel free to add more if you find any, or correct me when I'm wrong about Python not being needed at runtime (except for Python bindings)

- [ ] adios2
- [ ] cryptominisat
- [ ] csound
- [ ] fontforge
- [ ] gdal
- [ ] gdcm
- [ ] genometools
- [ ] geos
- [ ] glib
- [ ] gnuradio
- [ ] gobject-introspection
- [ ] gupnp
- [ ] halide
- [ ] itstool
- [ ] lcm
- [ ] ldns
- [ ] libpeas
- [ ] libtorrent-rasterbar
- [ ] lilypond
- [ ] mapserver
- [ ] notmuch
- [ ] ola
- [ ] opencolorio
- [ ] opencv
- [ ] opencv@3
- [ ] openimageio
- [ ] root
- [ ] simgrid
- [ ] soapysdr
- [ ] stp
- [ ] tbb
- [ ] tbb@2020
- [ ] uhd
- [ ] vapoursynth
- [ ] volk
- [ ] vtk
- [ ] vtk@8.2
- [ ] xapian

**Review by cho-m — COMMENTED** [review]
Unrelated to current discussion, but adding some review comments for some CI failures.

File: Formula/cython.rb [hunk 1]
```diff
@@ -18,20 +19,16 @@ class Cython < Formula
...
 
   def install
-    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
-    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
-    system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(libexec)
-
-    bin.install Dir[libexec/"bin/*"]
-    bin.env_script_all_files(libexec/"bin", PYTHONPATH: ENV["PYTHONPATH"])
+    # This is already keg-only, so we can install everything into the prefix.
+    system "python3", *Language::Python.setup_install_args(prefix)
```
[the remark this answers is not in the document]
Comment (cho-m) [reply Formula/cython.rb:26]: I wonder if there is a minor risk of de-sync'ing formulae if a user runs `pip install -U cython` or `pip install cython==<version>`.

Though, that would be a user issue based on Homebrew's stance for `python` formulae.

---

Anyway, I'm also fine with just linking it for an easier-to-manage formula.

**Pull request closed without merging**